### ISSUE TITLE
Style/HashSyntax incorrectly reporting mixed styles

### ISSUE THREAD

**Issue by twe4ked** [bbatsov/rubocop#3898]
Given the following input the `Syle/HashSyntax` cop incorrectly reports a mixed style error.

``` ruby
super(bar: bar, **args)
```

--------

## Expected behavior

No errors.

## Actual behavior

```
bar.rb:1:7: C: Style/HashSyntax: Don't mix styles in the same hash.
super(bar: bar, **args)
      ^^^^
```

## Steps to reproduce the problem

I tried adding a failing spec in `spec/rubocop/cop/style/hash_syntax_spec.rb`:

``` ruby
it 'does not register a mixed syntax offence' do
  inspect_source(cop, 'super(bar: bar, **args)'
  expect(cop.messages).to be_empty
end
```

But I couldn't reproduce it there.

I've created a simple repo that reproduces the error:

https://github.com/twe4ked/rubocop-mixed-hash-test

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
# test project
$ bundle exec rubocop -V
0.46.0 (using Parser 2.3.3.1, running on ruby 2.0.0 universal.x86_64-darwin15)

# actual project (same errors)
$ rubocop -V
0.46.0 (using Parser 2.3.3.1, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by Drenmi** [comment]
Good catch, @twe4ked. This is probably a regression caused by me, so I'll have a look. 😀

**Comment by Drenmi** [comment]
@twe4ked: It seems I had it the wrong way around, and this regression seems to be (accidentally) fixed by me introducing the `PairNode` refactor in current `master`. 😅

**Comment by twe4ked** [comment]
Thanks!

Updated https://github.com/twe4ked/rubocop-mixed-hash-test to use the `master` (c8d652dfa521526ff91cbf769e0d3551d50e1f3e) branch and I've confirmed it's fixed.

I'll leave this issue open until a new version is released. Feel free to close.

**Comment by Drenmi** [comment]
@twe4ked: 0.47 is now released. Feel free to close if you can assert this issue was fixed. 😀

**Comment by twe4ked** [comment]
Confirmed fixed in my project. Thanks all!
